src/MSIdentityScaffolding/Microsoft.DotNet.MSIdentity/MicrosoftIdentityPlatform/MicrosoftIdentityPlatformApplicationManager.cs

```diff
@@ -161,40 +162,84 @@ internal async Task<ApplicationParameters> CreateNewApp(TokenCredential tokenCre
-               .GetAsync()).First();
+                var existingApplication = (await graphServiceClient.Applications
+                   .Request()
+                   .Filter($"appId eq '{reconcialedApplicationParameters.ClientId}'")
+                   .GetAsync()).First();
 
-            // Updates the redirect URIs
-            var updatedApp = new Application
-            {
-                Web = existingApplication.Web
-            };
-            updatedApp.Web.RedirectUris = reconcialedApplicationParameters.WebRedirectUris;
+                // Updates the redirect URIs
+                if (existingApplication.Web == null)
+                {
+                    existingApplication.Web = new WebApplication();
+                }
 
-            // TODO: update other fields. 
-            // See https://github.com/jmprieur/app-provisonning-tool/issues/10
-            await graphServiceClient.Applications[existingApplication.Id]
-                .Request()
-                .UpdateAsync(updatedApp).ConfigureAwait(false);
+                var updatedApp = new Application
+                {
+                    Web = existingApplication.Web
+                };
+
+                //update redirect uris
+                List<string> redirectUris = updatedApp.Web.RedirectUris.ToList();
+                List<string> validUris = ValidateUris(toolOptions.RedirectUris).ToList();
+                redirectUris.AddRange(validUris);
+                updatedApp.Web.RedirectUris = redirectUris.Distinct();
 
-            if (existingApplication.RequiredResourceAccess != null
-                && !reconcialedApplicationParameters.IsBlazorWasm
-                && existingApplication.RequiredResourceAccess.Any()
-                && (existingApplication.PasswordCredentials == null
-                || !existingApplication.PasswordCredentials.Any()
-                || !existingApplication.PasswordCredentials.Any(password => !string.IsNullOrEmpty(password.SecretText))))
+                //update implicit grant settings
+                if (updatedApp.Web.ImplicitGrantSettings == null)
+                {
+                    updatedApp.Web.ImplicitGrantSettings = new ImplicitGrantSettings();
+                }
+
+                if (toolOptions.EnableAccessToken.HasValue)
+                {
+                    updatedApp.Web.ImplicitGrantSettings.EnableAccessTokenIssuance = toolOptions.EnableAccessToken.Value;
+                }
+
+                if (toolOptions.EnableIdToken.HasValue)
+                {
+                    updatedApp.Web.ImplicitGrantSettings.EnableIdTokenIssuance = toolOptions.EnableIdToken.Value;
+                }
+
+                // TODO: update other fields. 
+                // See https://github.com/jmprieur/app-provisonning-tool/issues/10
+                try
+                {
+                    await graphServiceClient.Applications[existingApplication.Id]
```

Comment:
Does this UpdateAsync return anything to indicate success/failure, or does it throw on error?

Reply:
throw on error.  am catching the exception currently and doing nothing with it. Will have a message for VS and CLI.